### DISCUSSION TITLE
How does split_into_segments work with data from excel files?

### DISCUSSION THREAD

**Discussion by ArminVarmaz** [KindOPSTAR/QualiGPT#2] [Q&A]
Dear all,

thank you for excelent research on using GPT and content analysis. I am using pythoin 3.9 on a mac and I updated to newest openai package, using openai migrate to migrate your chat-completition commands to the newer version. 

I am using an Excel-Files with two columns. First column has IDs and the second column social media posts. I would like to obtain key themes. However, I am constantely running into too many tokens problem. How does split_into_segments work with data from excel cells? 

Best
Armin

**Comment by KindOPSTAR** [comment]
Hi Armin,

Thank you for your interest in this project and for reporting this issue. After testing, I have identified the error and its cause, and I will fix it soon (possibly this week or next week!). What we know so far is that this error is more likely to occur with large datasets. I suggest that before this issue is fixed, you can use Python or other methods to split the dataset into smaller datasets. Currently, in my tests, datasets containing about 1000 social media posts are working fine. If you are using an extremely large social media dataset, please wait for my update! 

Thanks again!
Albert (QualiGPT Developer)

**Comment by ArminVarmaz** [comment]
Hi Albert,

thanks for pointing it out. Unfortunately, I experience the same issue again, even if I use only 500 posts. I attached my results txt-file, which may be helpful for you. It seems to me that there is communication to openai but not all segments are processed because the number of posts in each category does not sum up to 500. 

Armin
[Results_QualiGPT.txt](https://github.com/KindOPSTAR/QualiGPT/files/14021709/Results_QualiGPT.txt)

**Comment by KindOPSTAR** [comment]
Hi Armin,

Thanks for letting me know this! Sorry to hear you experienced this issue
again, I will do a test with your data sample this weekend, and will let
you know then. Sorry, I am too busy this week…

Best wishes,
Albert

**Comment by KindOPSTAR** [comment]
Hi Armion,

Thanks for waiting, just a friendly reminder, I've updated the QualiGPT to version v0.1.1-alpha. Although it might still not be perfect, the issues have finally been resolved.

Please note: Since QualiGPT is currently in a testing phase, it's important to set a usage cap for your API. Although the cost of GPT-3.5 Turbo is low, this could be a necessary action to reduce risk! 

Thank you for your support!

Best wishes,
Albert
